core/error.go

```diff
@@ -22,17 +22,40 @@ var (
 	// ErrKnownBlock is returned when a block to import is already known locally.
 	ErrKnownBlock = errors.New("block already known")
 
-	// ErrGasLimitReached is returned by the gas pool if the amount of gas required
-	// by a transaction is higher than what's left in the block.
-	ErrGasLimitReached = errors.New("gas limit reached")
-
 	// ErrBlacklistedHash is returned if a block to import is on the blacklist.
 	ErrBlacklistedHash = errors.New("blacklisted hash")
 
+	// ErrNoGenesis is returned when there is no Genesis Block.
+	ErrNoGenesis = errors.New("genesis not found in chain")
+)
+
+// State transition consensus errors, any of them encountered during
+// the block processing can lead to consensus issue.
```

Comment:
I'n not sure I understand what you mean by 'can lead to consensus issue'. Do you mean that if we hit this during evm-execution, some validation that should have occurred earlier must have failed?

Reply:
It's misleading, I change the comments.

My idea is making these errors explicit. Before execute a transaction/message-call, we will do lots of pre-check. If any pre-check clause failed, then the transaction is invalid which should never be packed. 

So it's a list of errors relevant to pre-checking.

Reply:
We can distinguish these errors with evm internal errors. Any error defined here can lead to "BAD block", while evm internal error is allowed during the state transition.